q2_fmt/_peds.py

```diff
@@ -466,11 +479,205 @@ def _create_masking(time_metadata, donor_df, recip_df, reference_column):
+                                                     reference_column)
+
+    if len(used_references.unique()) == 1:
+        raise AssertionError("There is only one donated microbiome in your"
+                             " data. A Monte Carlo simulation shuffles"
+                             " your donated microbiome and recipient pairing"
+                             " and needs more than one donated microbiome"
+                             " to successfully shuffle.")
+    recipient = metadata_df.loc[metadata_df[reference_column].notnull()]
+
+    if len(recipient.index.unique()) == 1:
+        raise AssertionError("There is only one recipient in your"
+                             " data. A Monte Carlo simulation shuffles"
+                             " your donated microbiome and recipient pairing"
+                             " and needs more than one recipient"
+                             " to successfully shuffle.")
+
+    peds = sample_peds(
+           table=table, metadata=metadata,
+           time_column=time_column,
+           reference_column=reference_column,
+           subject_column=subject_column,
+           filter_missing_references=filter_missing_references,
+           drop_incomplete_subjects=drop_incomplete_subjects,
+           drop_incomplete_timepoint=drop_incomplete_timepoint).set_index("id")
+    actual_temp = peds["measure"]
+
+    # Mismatch simulation:
+    table = table > 0
+    recip_df = _create_recipient_table(used_references, metadata_df, table)
+    donor_df = table[table.index.isin(used_references)]
+    duplicated_table, mismatched_df = \
+        _create_mismatched_pairs(recip_df,
+                                 metadata_df,
+                                 used_references,
+                                 reference_column)
+    donor_mask = _create_sim_masking(mismatched_df, donor_df, reference_column)
+    mismatched_pairs_n = len(duplicated_table.index)
+    recip_mask = donor_mask & duplicated_table
+    # Numerator for PEDS Calc. (Number of Donor features in the Recipient)
+    num_sum = np.sum(recip_mask.values, axis=1)
+    # Denominator for PEDS Calc. (Number of unique features in the Donor)
+    donor_sum = np.sum(donor_mask, axis=1)
+    # This ignores warnings that come from dividing by 0.
+    # mismatched_peds will be Nan if the denominator is 0 and thats reasonable.
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        mismatched_peds = num_sum/donor_sum
+    mismatchedpairs_df = _simulate_uniform_distro(recip_df, mismatched_peds,
+                                                  iterations)
+    per_sub_stats = _per_subject_stats(actual_temp, mismatchedpairs_df,
+                                       iterations, mismatched_pairs_n)
+    global_stats = _global_stats(per_sub_stats['p-value'])
+    return per_sub_stats, global_stats
+
+
+# peds_simulation helper functions
+
+def _create_mismatched_pairs(recip_df, metadata, used_references,
```

Comment:
This function going to be problematic for memory usage with large data sets, since you're duplicating recipient sample vectors in the feature table. This should ultimately be refactored such that the recipient sample vectors don't need to be duplicated (which will involve changes to how you're actually computing PEDS). 

It's ok to create an issue for this, rather than address as part of this pull request.

Reply:
This is in reference to `_create_duplicated_recip_table` (which was previously part of this function).